packages/dashmate/src/listr/tasks/setup/regular/configureSSLCertificateTaskFactory.js

```diff
@@ -61,6 +66,34 @@ export default function configureSSLCertificateTaskFactory(
                   return 'the same path for both files';
                 }
 
+                const bundlePem = fs.readFileSync(chainFilePath, 'utf8');
+                const privateKeyPem = fs.readFileSync(privateFilePath, 'utf8');
+
+                // Step 2: Create a signature using the private key
+                const data = 'This is a test message';
+                const sign = crypto.createSign('SHA256');
+                sign.update(data);
+                sign.end();
+
+                const signature = sign.sign(privateKeyPem, 'hex');
```

Comment:
it should be enough to compare public keys. 

Not sure about the library you use, but sth around the lines:

```
privateKeyPem.publicKey() == certificate.publicKey()
```

Reply:
I've tried something like that:

```js

// Paths to your files
const privateKeyPath = '/path/to/private.key';
const bundlePath = '/path/to/bundle.crt';

// Step 1: Read the private key and bundle
const privateKeyPem = readFile(privateKeyPath);
const bundlePem = readFile(bundlePath);

// Step 2: Create the public key from the private key
const privateKey = crypto.createPrivateKey(privateKeyPem);
const publicKeyFromPrivateKey = privateKey.export({
  type: 'spki',
  format: 'pem',
});

// Step 3: Extract the public key from the certificate in the bundle
// Assume that the first certificate in the bundle is the server certificate
const certificate = crypto.createPublicKey(bundlePem);
const publicKeyFromCertificate = certificate.export({
  type: 'spki',
  format: 'pem',
});

// Step 4: Compare the public keys
if (publicKeyFromPrivateKey === publicKeyFromCertificate) {
  console.log('The private key matches the certificate.');
} else {
  console.log('The private key does NOT match the certificate.');
}
```

but it fails with:

```
Error: error:1E08010C:DECODER routines::unsupported
    at Object.createPrivateKey (node:internal/crypto/keys:632:12)
    at FormPrompt.validate (file:///Users/ivanshumkov/Projects/dashpay/platform/packages/dashmate/src/listr/tasks/setup/regular/configureSSLCertificateTaskFactory.js:71:43)
    at FormPrompt.submit (/Users/ivanshumkov/Projects/dashpay/platform/.yarn/cache/enquirer-https-9a17ef3ad5-390380d1a9.zip/node_modules/enquirer/lib/prompt.js:123:49)
    at FormPrompt.submit (/Users/ivanshumkov/Projects/dashpay/platform/.yarn/cache/enquirer-https-9a17ef3ad5-390380d1a9.zip/node_modules/enquirer/lib/prompts/form.js:191:30)
    at FormPrompt.keypress (/Users/ivanshumkov/Projects/dashpay/platform/.yarn/cache/enquirer-https-9a17ef3ad5-390380d1a9.zip/node_modules/enquirer/lib/prompt.js:44:23)
    at ReadStream.on (/Users/ivanshumkov/Projects/dashpay/platform/.yarn/cache/enquirer-https-9a17ef3ad5-390380d1a9.zip/node_modules/enquirer/lib/keypress.js:205:26)
    at ReadStream.emit (node:events:530:35)
    at emitKeys (node:internal/readline/utils:371:14)
    at emitKeys.next (<anonymous>)
    at ReadStream.onData (node:internal/readline/emitKeypressEvents:64:36)
    at ReadStream.emit (node:events:518:28)
    at addChunk (node:internal/streams/readable:559:12)
    at readableAddChunkPushByteMode (node:internal/streams/readable:510:3)
    at Readable.push (node:internal/streams/readable:390:5)
    at TTY.onStreamRead (node:internal/stream_base_commons:190:23) {
  library: 'DECODER routines',
  reason: 'unsupported',
  code: 'ERR_OSSL_UNSUPPORTED'
}
```